### DISCUSSION TITLE
When pushing a commit git asks for username and password in Linux

### DISCUSSION THREAD

**Discussion by arsekil** [community/community#94212] [New to GitHub]
Hello,

I don';t understand why is this happening. Every time I am pushing a commit in Linux in VSCode git asks me for username and password.

**Comment by NipulM** [comment]
Hello 👋 

On Linux, Git might be configured to use a credential manager that does not store your credentials.
Use a credential manager that can store credentials. You can configure Git to use the credential manager by doing:
`git config --global credential.helper cache
`
By default this will cache your credentials for 15 minutes

Make sure your Git configuration has the correct username and email set.
Run the following commands to set them up:

```
git config --global user.name "your_name"
git config --global user.email "your_email"
```

Hope this helps!

**Comment by khot-aditya** [comment]
@andrejmoltok you can try what @NipulM is suggesting or simply use SSH to pull/push or commit.
You can either add your machine's SSH key to account or use ssh@github.com URL method.

**Comment by sonersyln** [comment]
Open Terminal.

Enter the following command:

git config --global credential.helper store
Now, when you push a commit, Git will ask for your username and password one last time. Enter them, and they should be saved for future use.
